### ISSUE TITLE
Can anyone reproduce Table.1.(a) in paper?

### ISSUE THREAD

**Issue by Luciennnnnnn** [Fanghua-Yu/SUPIR#151]
I have tried to reproduce results of SUPIR in RealPhoto60 with pretrained model provided by authors, however the number is much lower than ones in paper:
My:
```Bash
clipiqa maniqa musiq 
0.5697 0.3584 59.7698
```
Paper:
```Bash
clipiqa maniqa musiq 
0.8232 0.4295 73.00
```

**Comment by CruelBrutalMan** [comment]
@Luciennnnnnn 
I didn't try to repeat the results in the article, but what you found may indicate a violation of journalistic ethics, when authors state unverified facts in their research. Well, we can assume that many articles are published in modern times - with unverified metrics

**Comment by FurkanGozukara** [comment]
SUPIR is still the very best image upscaler even though it has been like 1 year you know that right?

imagine 1 year in AI and still unbeaten and unchallenged

**Comment by ceteke** [comment]
I also get the same results as @Luciennnnnnn when I try to replicate the results reported in the paper.  Did you manage to replicate somehow?

**Comment by JasonGUTU** [comment]
SUPIR can only achieve better results than the paper. Hyperparameters are very important, so be sure to tune them.

**Comment by Luciennnnnnn** [comment]
Hi @JasonGUTU , If hyperparameters matters, could you please provide "correct" hyperparameters?

**Comment by CruelBrutalMan** [comment]
> Hi [@JasonGUTU](https://github.com/JasonGUTU) , If hyperparameters matters, could you please provide "correct" hyperparameters?

I completely agree with @Luciennnnnnn! @JasonGUTU just don't say that for each case you need to select your own parameters based on the features of the image, etc. Please provide the settings with which we can get the same result as in the article.

**Comment by JasonGUTU** [comment]
There are four important points I'd like to clarify:

First, SUPIR is not an end-to-end model; instead, it's an interactive software framework with numerous configurable options. To achieve optimal results, hyperparameters should be selected based on specific image conditions. I  believe that simply adjusting these parameters can significantly enhance image quality on the testset. I recommend closely inspecting the images you've obtained. If there are inherent quality issues with your images, we would be more than happy to discuss this further.

Second, standard Image Quality Assessment (IQA) metrics generally have limited value for evaluating models like SUPIR. Extensive research by our team has consistently demonstrated that virtually no IQA metric reliably correlates with perceptually-oriented image quality. In fact, higher numerical values often correspond to worse visual results and that is a conclusion even since 2018. The numerical results published in our paper were derived from the initial version of SUPIR (obtained in late October 2023, we use it to write the paper). The publicly available version, trained in February 2024 on a significantly larger computational cluster, far exceeds the performance of the paper version. 

Although it's possible that certain parameter configurations might maximize SUPIR’s numeric performance on specific datasets, finding such configurations is  meaningless because optimizing metrics does not align with the intended use of perceptually-oriented models. Furthermore, better numerical scores do not necessarily indicate better visual quality. We have deliberately chosen not to optimize our models solely for arbitrary metrics or expend effort on only numbers. All subsequent models developed after SUPIR no longer reference IQA metrics.

Third, SUPIR's success is widely recognized, demonstrated by its extensive commercial adoption across various fields. We have since updated SUPIR with two subsequent generations. Importantly, SUPIR’s success has never depended on IQA metrics but rather on broad recognition and positive community feedback.

Fourth, while open-sourcing our models is not an obligation, we have  made SUPIR publicly available. Everyone is encouraged to test it on their own images. We welcome fair comparisons of SUPIR in any research publications, as facilitating such comparisons was our primary motivation for releasing the model.

**Comment by JasonGUTU** [comment]
Returning to your original question: The parameters provided in `test.py` ([https://github.com/Fanghua-Yu/SUPIR/blob/master/test.py](https://github.com/Fanghua-Yu/SUPIR/blob/master/test.py)) are placeholders intended only as a starting point. The core nature of the SUPIR model requires manual parameter tuning and experimentation to achieve optimal results. We provide a Gradio-based interactive interface to help you easily select and test appropriate parameter values.

Here are some recommended empirical parameters based on typical usage:

* **Stage 2 adjustment**:

  * Option 0: `s_stage2 = 1.0`
  * Option 1: `s_stage2 = 0.93`

* **Quality-oriented settings** (higher visual quality, potentially lower fidelity):

  * `s_cfg = 6.0`, `spt_linear_CFG = 3.0`, `s_noise = 1.02`

* **Fidelity-oriented settings** (closer to original input fidelity, may sacrifice some visual appeal):

  * `s_cfg = 4.0`, `spt_linear_CFG = 1.0`, `s_noise = 1.01`

The parameters listed above generally have significant impact on the final image quality and appearance.

**Comment by JasonGUTU** [comment]
> > We welcome fair comparisons of SUPIR in any research publications, as facilitating such comparisons was our primary motivation for releasing the model.
> 
> but how can we make fair comparisons if we can't achieve the result you have in the original article? Honestly making comparisons [@Luciennnnnnn](https://github.com/Luciennnnnnn) got a result that is several times worse than in your article. And therefore, a reasonable question - how to make a comparison if in fact the metrics show results that are much worse than they should be? no article will be published if it contains metric values ​​that are much worse than in the original

Yes Exactly. However, [@Luciennnnnnn](https://github.com/Luciennnnnnn) only used the placeholder parameters provided directly in the code. At the very least, several parameter combinations should be tested to achieve better performance. I believe this is the core principle of a fair comparison: If your competitors could achieve a score of 100, but you don't fully explore your parameter space—whether due to a lack of information or deliberate choice—this essentially constitutes unfair competition.

This reasoning explains precisely why we didn't offer default parameters and even provided unrealistic placeholder parameters—to emphasize the critical importance of careful parameter selection. Most users of our codebase are enthusiasts or professional researchers who explicitly understand that parameter tuning is essential and that no universally optimal parameter setting exists.

We've observed multiple cases where the performance of SUPIR has been intentionally underreported in some papers. Simply selecting slightly inappropriate parameters is enough to artificially degrade its results. But this isn't our business.

**Comment by JasonGUTU** [comment]
Therefore, what I’d like to emphasize is this: **SUPIR is fundamentally a tool offering numerous possibilities. It should not be treated as an out-of-the-box, fully automated algorithm, as we have never claimed it to be so.**

However, we understand the importance of clarity and will update the README to explicitly highlight this point. Additionally, we'll provide a set of more reasonable and broadly applicable default parameters in `test.py` as a starting point, replacing the current intentionally exaggerated placeholders.

**Comment by ceteke** [comment]
I do understand and can also validate the "meaninglessness" of the IQA metrics. They do not really reflect what is going on in an image, and I also understand that hyperparameters can be fine-tuned and changed (sometimes deliberately) to make the results look worse. However, if you publish results at big conferences, there should be more transparency. At least for the very common benchmark datasets like DIV2K. You cannot just say "Adjust the hyperparameters" and claim that some researchers are "deliberately" using wrong hyperparameters to make your results look worse. If you lack transparency, how can you expect a fair comparison? How do we then know that (this is by no means an accusation) you did not deliberately make the results look better? Nonetheless, thank you for the further clarification. I will also try out the suggested hyperparameters.
